simba/schedule.py

```diff
@@ -285,10 +301,170 @@ def assign_vehicles(self):
+                    # or station is not electrified. in this case vehicles should not strand there
+                    # even if rotations are not possible
+                    rot.vehicle_id = vehicle_id
+                    all_standing_vehicles.remove((vehicle_id, depot))
+                    vehicle_data[vehicle_id] = {"soc": end_soc,
+                                                "arrival_time": rot.arrival_time}
+                    break
+            else:
+                # no vehicle available for dispatch, generate new one
+                vehicle_type_counts[vt_ct] += 1
+                vehicle_id = f"{vt_ct}_{vehicle_type_counts[vt_ct]}"
+                rot.vehicle_id = vehicle_id
+                end_soc = initial_soc - consumption_soc
+                vehicle_data[vehicle_id] = {"soc": end_soc,
+                                            "arrival_time": rot.arrival_time}
+
+            rotations_in_progress.append(rot)
+            try:
+                next_rot = rotations[k + 1]
+                departure_time = next_rot.departure_time
+            except IndexError:
+                pass
+            else:
+                # keep list of rotations in progress sorted by soc reached at the time of the next
+                # departure.
+                rotations_in_progress = sorted(rotations_in_progress, key=lambda x: x.arrival_time)
+
+        # update vehicle ID for nice natural sorting
+        for rot in rotations:
+            # get old vehicle id (vehicle type + charging type, sequential number)
+            vt_ct, old_num = rot.vehicle_id.rsplit('_', 1)
+            # how many vehicles of this type?
+            vt_cnt = vehicle_type_counts[vt_ct]
+            # easy log10 to get max needed number of digits
+            digits = len(str(vt_cnt))
+            # assign new zero-padded ID (all of same vehicle type have same length)
+            rot.vehicle_id = f"{vt_ct}_{int(old_num):0{digits}}"
+        self.vehicle_type_counts = vehicle_type_counts
+
+    def get_charge_curves(self, charge_levels, final_value: float, time_step_min: float) -> dict:
+        """ Get the numeric charge curves.
+
+        :param charge_levels: different power levels which clip the charge curves
+        :type charge_levels: set[float]
+        :param final_value: soc value at which the numeric calculation stops
+        :type final_value: float
+        :param time_step_min: time_step in minutes for the numeric calculation
+        :type time_step_min: float
+        :return: soc over time curves
+        :rtype: dict
+        """
+        charge_curves = dict()
+        for vehicle_name, vehicle_type in self.vehicle_types.items():
+            charge_curves[vehicle_name] = dict()
+            for ct_name, v_info in vehicle_type.items():
+                charge_curves[vehicle_name][ct_name] = dict()
+                default_eff = Battery(1, [[0, 1], [1, 1]], 0).efficiency
```

Comment:
probably default efficiency of battery vehicle is more interesting. Or build a SpiceEV.component.VehicleType from v_type and read out actual efficiency